javascript/javascript3/week3/promises/main.js

```diff
@@ -0,0 +1,47 @@
+// https://api.github.com/search/repositories?q=user:benna100
+// https://api.github.com/search/repositories?q=user:panivita
+// https://api.github.com/search/repositories?q=user:LucyChyzhova
+
+const username = ["benna100", "panivita", "LucyChyzhova"];
+
+// Lets use the github api to see what repositories different users have.
+const username1 = "benna100";
+const username2 = "panivita";
+const username3 = "LucyChyzhova";
+
+const promise1 = fetch(
+  `https://api.github.com/search/repositories?q=user:${username1}`
+);
+
+const promise2 = fetch(
+  `https://api.github.com/search/repositories?q=user:${username2}`
+);
+
+const promise3 = fetch(
+  `https://api.github.com/search/repositories?q=user:${username3}`
+);
+
+Promise.all([promise1, promise2, promise3])
+  .then((responses) =>
+    Promise.all(responses.map((response) => response.json()))
+  )
```

Comment:
Well done adding names to a URL, but on the lines 24-26 the code is a bit overcomplicated. 
The best way is probably to write a function to fetch data. But it's just a suggestion.

Reply:
Hi,
Can you check again ? I have written a function to fetch the data
thank you.